src/context/CartContext.jsx

```diff
@@ -30,13 +52,66 @@ function CartProvider({ children }) {
     }
   }
 
+  //
+
+  function incrementHandler(id) {
+    const isItemInCart = cart.find((cartItem) => cartItem.id === id); // check if the item is already in the cart
+
+    if (isItemInCart) {
+      setCart(
+        cart.map(
+          (
+            cartItem, // if the item is already in the cart, increase the quantity of the item
+          ) =>
+            cartItem.id === id
+              ? { ...cartItem, quantity: cartItem.quantity + 1 }
+              : cartItem, // otherwise, return the cart item
+          // console.log(cart),
+        ),
+      );
+    } else {
+      // setCart([...cart, { ...item, quantity: 1 }]); // if the item is not in the cart, add the item to the cart
+    }
+  }
+
+  // ////////
+
+  function decrementHandler(id) {
+    const isItemInCart = cart.find((cartItem) => cartItem.id === id); // check if the item is already in the cart
+
+    if (isItemInCart) {
+      const newCartData = cart.map(
+        (
+          cartItem, // if the item is already in the cart, increase the quantity of the item
+        ) =>
+          cartItem.id === id
+            ? {
+                ...cartItem,
+                quantity:
+                  cartItem.quantity > 1
+                    ? cartItem.quantity - 1
+                    : cartItem.quantity,
+              }
+            : cartItem, // otherwise, return the cart item
+        // console.log(cart),
+      );
```

Comment:
```suggestion
      const newCartData = cart.map(
        (
          cartItem, // if the item is already in the cart, increase the quantity of the item
        ) =>
          cartItem.id === id
            ? {
                ...cartItem,
                quantity:
                  cartItem.quantity > 1
                    ? cartItem.quantity - 1
                    : cartItem.quantity,
              }
            : cartItem, // otherwise, return the cart item
        // console.log(cart),
      );
      const newCartData = cart.map((cartItem) => {
  if (cartItem.id === id) {
    if (cartItem.quantity > 1) {
      return {
        ...cartItem,
        quantity: cartItem.quantity - 1,
      };
    } else {
      return {
        ...cartItem,
        quantity: cartItem.quantity,
      };
    }
  } else {
    return cartItem;
  }
});

```